src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { ReactElement, useCallback, useEffect, useMemo } from 'react'
+import { CircularProgress, Typography } from '@mui/material'
+import { trackSafeAppOpenCount } from '@/services/safe-apps/track-app-usage-count'
+import { useSafeAppFromManifest } from '@/hooks/safe-apps/useSafeAppFromManifest'
+import { useSafeApps } from '@/hooks/safe-apps/useSafeApps'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { isSameUrl } from '@/utils/url'
+import { ThirdPartyCookiesWarning } from './ThirdPartyCookiesWarning'
+import useThirdPartyCookies from './useThirdPartyCookies'
+import useAppIsLoading from './useAppIsLoading'
+
+import css from './styles.module.css'
+
+export type TransactionParams = {
+  safeTxGas?: number
+}
+
+type AppFrameProps = {
+  appUrl: string
+}
+
+const AppFrame = ({ appUrl }: AppFrameProps): ReactElement => {
+  const { safe } = useSafeInfo()
+  const { remoteSafeApps } = useSafeApps()
+  const { safeApp: safeAppFromManifest } = useSafeAppFromManifest(appUrl, safe.chainId)
+  const { thirdPartyCookiesDisabled, setThirdPartyCookiesDisabled } = useThirdPartyCookies()
+  const { iframeRef, appIsLoading, isLoadingSlow, setAppIsLoading } = useAppIsLoading()
+
+  const currentApp = useMemo(
```

Comment:
This will not work for custom apps, we need to fetch the manifest here

Reply:
Ahh I see that it's need because we use the id for tracking then I'd create a new hook and fetch a single app instead of bringing the whole list

Reply:
Ah, true, the backend supports an id param, doesn't it.

Reply:
`useCustomSafeApps` retrieve all the manifests so perhaps we can add a `getCustomApp` method to `useSafeApps`. I like to have the hook to retrieve only one manifest but in this case we are using both hooks so doesn't make sense and even the network request is duplicated

Reply:
Also the list is going to be already http-cached, so it’s actually faster the current way 👍

Reply:
It's not cached, unfortunately, there are no caching headers for this request, already asked the infra team why. 

Also, this approach won't work if the list gets paginated.

Reply:
made some changes for avoid the retrieval of all the custom apps manifests when entering the web directly with the param `?appUrl=`

Reply:
Good point, Mikhail

Reply:
> made some changes for avoid the retrieval of all the custom apps manifests when entering the web directly with the param `?appUrl=`

I am sorry, but I don't understand how it helps with my outlined problem. When you enter the app frame, it fetches the whole list of apps, and then it searches the app on the client. What it could do instead is fetch the app by the URL from the backend to avoid fetching the whole list

Reply:
How can i get the app by URL?

Cannot see the method https://safe.global/safe-react-gateway-sdk/modules.html

Reply:
What I improved is the manifest retrieval for avoid multiple requests. If we have a method for fetch a single app i can use it no problem

Reply:
Aaah i see it https://safe.global/safe-react-gateway-sdk/functions/getSafeApps.html

Reply:
a query param :D

Reply:
The gateway doesn't expose the `url` param and has a code freeze at the moment, so we decided to hold on to this change